libtransmission/announcer-udp.cc

```diff
@@ -291,7 +291,7 @@ struct tau_tracker
 {
     using Mediator = tr_announcer_udp::Mediator;
 
-    tau_tracker(Mediator& mediator, tr_interned_string key_in, tr_interned_string host_in, tr_port port_in)
+    tau_tracker(Mediator& mediator, tr_interned_string const& key_in, tr_interned_string const& host_in, tr_port port_in)
```

Comment:
I would question whether or not this gives a benefit? `tr_interned_string` is a  `std::string_view` + a `size_t`. I would expect all the reasons to pass a `std::string_view` by value hold here too?

Some background reading: https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/

Reply:
Thanks for the background info.

Another quote:

"Things (aggregates, objects, arrays, data structures) that are sufficiently large in memory footprint should always be designed to facilitate pass-by-reference, for performance reasons. This advice definitely applies when it is **hundreds of bytes** or more. This advice is borderline when it is tens of bytes."
 https://softwareengineering.stackexchange.com/questions/372105/is-passing-arguments-as-const-references-premature-optimization

Reply:
...hey, looks like you then went and found a few places in the code that had `interned_string const&` and fixed them to pass by value. Thanks! :tada: